libdevcore/OverlayDB.cpp

```diff
@@ -129,8 +129,16 @@ void OverlayDB::rollback()
 std::string OverlayDB::lookup(h256 const& _h) const
 {
     std::string ret = MemoryDB::lookup(_h);
-    if (ret.empty() && m_db)
-        DEV_IGNORE_EXCEPTIONS(ret = m_db->lookup(toSlice(_h)));
+    if (!ret.empty() || !m_db)
+        return ret;
+
+    try
+    {
+        ret = m_db->lookup(toSlice(_h));
+    }
+    catch (db::NotFound const&)
```

Comment:
Is there any place where this exception is useful? Maybe lookup should return just an empty output (or optional).
Also, not finding an entry in a database is not an exceptional case (especially when the method is called "lookup", not "get").

Reply:
I think I'll try to get rid of `NotFound` exception